### DISCUSSION TITLE
`default_factory` of pydantic's `Field` not used when included in `response_model`

### DISCUSSION THREAD

**Discussion by fzembow** [fastapi/fastapi#8853] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

# Reproduction

```Python
from fastapi import FastAPI
from pydantic import BaseModel, Field

app = FastAPI()

class TestModel(BaseModel):
    str_a: str = "hello world"
    str_b: str = Field(default_factory=lambda: "hello world")
    list_a: List[str] = []
    list_b: List[str] = Field(default_factory=list)


@app.get("/", response_model=TestModel)
def test():
    m = TestModel()
    # Defaults are properly added to the model...
    assert m.str_a == "hello world"
    assert m.str_b == "hello world"
    assert type(m.list_a) == list
    assert type(m.list_b) == list

    # But when generated due to a `response_model`, they are `null`
    return {}
```

### Description

* Open the browser and call the endpoint `/`.
* It returns a JSON with `{"str_a":"hello world","str_b":null,"list_a":[],"list_b":null}`: There are `null` instead of the `[]` that `default_factory` would provide.
* But I expected it to return `{"str_a":"hello world","str_b":hello world,"list_a":[],"list_b":[]}`.
* Note that if I instantiate the pydantic model directly, as I do within the handler itself, the defaults are correctly created.

### Environment

* OS: mac OS
* FastAPI Version: 0.64.0
* Python version: 3.9.1

### Context

* The reason I am doing this is that I want to put `example` within the `Field` as well as have a default, so I can't use the `=` as in the simple case.
* A workaround of course is to just return the manually-instantiated model in the handler - but that feels error prone and 
* Is this because [`default_factory`](https://pydantic-docs.helpmanual.io/usage/models/#field-with-dynamic-default-value) is still technically in beta? I'd be happy to implement support for it if so

**Comment by FalseDev** [comment]
I think the `response_model` kwarg is used for documentation only and isn't considered anywhere else

**Comment by Hultner** [comment]
Seems a bit unexpected for me considering list_b and str_b aren't Optional/nullable, I'd expect an either internal validation error causing a 500 or the expected return value as described in the post, i.e. with `hello_world` and `[]`.

I'd be surprised to see this, wouldn't it break the contract given by the OpenAPI schema?
To me this looks more like a bug. Either that or that it's intentionally not implemented due to the beta status, but then it would be nice to raise a not implemented error and not silently create data that isn't allowed by the model.

The obvious workaround would be to use a validator-function instead to instantiate the dynamic default value, this is what I usually do.

**Comment by fzembow** [comment]
@FalseDev in the documentation for [Response Model](https://fastapi.tiangolo.com/tutorial/response-model/#response-model) it's explicitly stated that 

> FastAPI will use this response_model to:
> Convert the output data to its type declaration.

So I'd expect it to in fact do that :)

@Hultner yeah it definitely seems like a bug... by setting a default I'm explicitly indicating that I do not want `null` there :)

I found this quite surprising as well, I'll look into the implementation in the next few days to understand it.

**Comment by usrbinsam** [comment]
https://github.com/tiangolo/fastapi/blob/bee35f5ae1fc58e7ab125427ad4287210e99d8b3/fastapi/utils.py#L94-L103 This appears to be why default_factory is not supported - it is not copied into the cloned response model.

Locally, I added `new_field.default_factory = field.default_factory` under line 98 and this appeared to fix the issue. Not sure if that's "enough" for a PR though.

**Reply by vvanglro** [reply]
This works for me.

**Comment by YuriiMotov** [comment]
Works fine with Pydantic V2, but still doesn't work with Pydantic V1
